src/Addon.ts

```diff
@@ -0,0 +1,111 @@
+import { Request } from 'express';
+
+import { AuthError, AuthErrorCodes } from './AuthError';
+import { CustomTokenExtractor, Products, TokenPayload } from './types';
+import * as util from './util';
+
+export interface Credentials {
+  sharedSecret: string;
+}
```

Comment:
Object with one property may be replaced with the value of that single property. It would break compatibility, but improve API.